kubernetes/kubernetes_service.go

```diff
@@ -80,10 +83,8 @@ func (in *IstioClient) GetServiceDetails(namespace string, serviceName string) (
 	}()
 
 	go func() {
-		deployments, err := in.k8s.AppsV1beta1().Deployments(namespace).List(emptyListOptions)
-		deployments = filterDeploymentsByService(serviceName, deployments)
+		deployments, err := in.k8s.AppsV1beta1().Deployments(namespace).List(*getDeploymentFilterListOptions(serviceName))
```

Comment:
This is good.
I have a minor suggestion (non blocker).
When using pointers, instead to apply to the function like
```
*getDeploymentFilterListOptions(serviceName)
```
I see more clear to use a aux variable, at least for me it helps in terms of readability.